apps/zotonic_launcher/bin/zotonic

```diff
@@ -17,6 +17,15 @@ export ZOTONIC_BIN
 cd -- "${ZOTONIC}" || \exit 1
 
 case "$1" in
+    completion)
+        COMMAND_NAMES="$(${ZOTONIC_BIN}/zotonic.escript command_names)"
+        # TODO: Check OS (this works in Ubuntu 21.10)
+        SYS_COMPLETION_DIR="/etc/bash_completion.d"
+        OUTPUT="$SYS_COMPLETION_DIR/zotonic-completion.bash"
+        # TODO: Completion for options
+        sudo sh -c "printf '#/usr/bin/env bash\n\n_zotonic_completions()\n{\n\tif [ \"\${#COMP_WORDS[@]}\" != \"2\" ]; then\n\t\treturn\n\tfi\n\n\tCOMPREPLY=(\$(compgen -W %s \"\${COMP_WORDS[1]}\"))\n}\n\ncomplete -F _zotonic_completions zotonic' '$COMMAND_NAMES' >| '$OUTPUT'"
+        sudo bash -c "source $OUTPUT"
```

Comment:
Is sudo this really necessary?

Reply:
I suppose it was in previous commits what was changed and `sudo` was removed.

Reply:
Yep, it was change at https://github.com/zotonic/zotonic/pull/2976/commits/c1f71067627b1720759b58e0f9954734f5f35323